diff --git a/src/dayz-linux-launcher/mainwindow.cpp b/src/dayz-linux-launcher/mainwindow.cpp
index d9223db..8362803 100644
--- a/src/dayz-linux-launcher/mainwindow.cpp
+++ b/src/dayz-linux-launcher/mainwindow.cpp
@@ -719,7 +719,7 @@ void MainWindow::load_mods_from_html(std::string const &path)
         }
         catch (std::exception const &e)
         {
-            spdlog::warn("Failed adding apparently existing mod '{}', reason: {}", it->at("path"), e.what());
+            spdlog::warn("Failed adding apparently existing mod '{}', reason: {}", it->at("path").get<std::string>(), e.what());
         }
     }
 
